### ISSUE TITLE
ios build.json debug/release signing not working as i expected

### ISSUE THREAD

**Issue by godexsoft** [apache/cordova-ios#507]
Creating a new issue for this as I've been told to do.

The problem is that in Xcode there is two sections for signing, one fro DEBUG and one for RELEASE. 
Typically you would have your developer profile setup for debug so that you can build and run a debug build on your device/simulator. And you would also have the release section filled in with either ad-hoc or appstore profile to be able to hit archive and create a release build that can be then uploaded to testflight/hockeyapp/appstore etc.

What i'm expecting is that in json.build i can specify debug and release sections and they will fill in the corresponding xcode sections.. however the reality is that only either one is used depending whether i pass --debug or --release to cordova CLI (i.e. cordova build --device ios --debug).

So basically the result is either i have my development profile in both debug and release on xcode and it does not work because i can't have development profile in release anyway. OR i end up having ad-hoc in both debug and release which of course also does not work for the same reason :)

Note: i'm using old buildsystem (--buildFlag="-UseModernBuildSystem=0").

Please do let me know if i'm using it in a wrong way. But i tried many combinations and it looks like it is what it is. I understand it may be the "expected" outcome in cordova terms as i'm clearly specifying --release or --debug in CLI but it does not make any sense in xcode terms so it should be fixed i think.

Thanks in advance.

**Comment by janpio** [comment]
Do the builds with `--release` and `--debug` work as expected?

If so, that is probably all that was intended by this feature when it was created. The `build.json` file was probably introduced as an alternative way to supplying the certificate information via the command line - and there is probably also configures all the settings to the one supplied data.

But, of course you are not wrong that it could be nice to have a "better" configured Xcode project. Someone would have to take a look at the code to see if this can be refactored to work that way, and check especially was happens when only one of both are supplied via `build.json`.

**Comment by godexsoft** [comment]
I am already using --release and --debug.. that's exactly the issue. 
Even tho i was able to get debug build to work fully from CLI when i specify automaticProvisioning=true and let xcode figure out which profile to use but that only will work for "debug" in xcode terms.

Building without --release nor --debug seems to have the same problem (CLI is assuming debug automatically).

I imagine there may be a way to get this to work thru other means but i was quite happy to see one can just use json.build for both android and iOS. Would be great to fix.

**Comment by janpio** [comment]
So both `--release` and `--debug` properly get the information configured in `build.json` and successfully create the specified type of build, correct? I just want to make sure that everything documented works as I think it does.

The problem is just, that after using one of these options, the changes to the Xcode project are not what you expected them to be and what they optimally would be to enable using the Xcode project directly in Xcode, correct?

**Comment by godexsoft** [comment]
ah, i see what you mean. no, unfortunately not. the problem is that xcodebuild complains about the mismatched provisioning profile/certificate for the other build variant (debug if release selected and vica versa). that's probably something relatively new in newer xcode and i imagine it was not the case before.

I was able to get the build to work just fine with --noSign option but that completely omits signing and exporting the .ipa file so it's no good. 

Currently if i use CLI, even tho the described problem is there when i open the xcode project in xcode I can select the correct profiles and it will build just fine. So signing is the only issue.

**Comment by godexsoft** [comment]
Any progress on this?

**Comment by janpio** [comment]
I can't reproduce this.

Is this reproducible in a new, plain Cordova app? A minimal reproduction repository would really help to debug and later fix this issue. More information on how to create one: https://github.com/apache/cordova-contribute/blob/master/create-reproduction.md

**Comment by dpogue** [comment]
I've marked this issue as stale because it's been several years with no further comments. If this is still an issue in the latest cordova-ios version and an up-to-date Xcode version, please let us know. Otherwise, this issue will be closed.

**Comment by dpogue** [comment]
Closing as stale.
